### PR TITLE
Add hidden to generated class context

### PR DESCRIPTION
**Pull request by sarangan12**
This PR is to address the issue reported at https://github.com/Azure/autorest.typescript/issues/832. A detailed description has been provided in the issue. 

In this PR, I have done the following changes:

1. Updated ```ts-morph``` to version ```9.0.0``` from ```5.0.0```. One Year back, when this project was created, the latest version of ```ts-morph``` was ```5.0.0```. It did not have the ability to add tags which is what we are doing in this PR. That feature was introduced later. So, I upgraded it to latest version. 

2. With the latest upgrade to ```ts-morph``` the documentation comments (with only one line) have changed to one line format from multi-line format. 

3. I have introduced the option ```has-custom-layer```. In this PR, this option is used to determine if we need to hide the Client and Client Context classes. In the future, we could extend its usage.

@joheredi @deyaaeldeen Please review and approve.

### CODE REVIEW

File: src/generators/clientContextFileGenerator.ts [hunk 1]
```diff
@@ -20,7 +20,8 @@ import { formatJsDocParam } from "./utils/parameterUtils";
 export function generateClientContext(
   clientDetails: ClientDetails,
   packageDetails: PackageDetails,
-  project: Project
+  project: Project,
+  hasCustomLayer: boolean
```
Comment (deyaaeldeen) [inline src/generators/clientContextFileGenerator.ts:24]: Perhaps the name is a bit unclear. One alternative could be `exportedByPackage`. Either way, please document it.

**Review by deyaaeldeen — COMMENTED** [review]
Looks reasonable to me but I am not sure why comments changed from:
```
/**
 *
 */
```
to:
`/** */`

Is that because of the upgrade to ts-morph 9?

File: src/generators/clientContextFileGenerator.ts (hunk 1 above)
Comment (sarangan12) [reply src/generators/clientContextFileGenerator.ts:24]: The ```hasCustomLayer``` could be used for multiple purposes in the future. Not just hiding classes. Added some documentation for the property.

**Review by joheredi — APPROVED** [review]
Looks good! Thanks for fixing this, as we discussed offline the only comment is to use the flag name `hideClients` instead.

**Pull request merged**